Crashing after switch to subprocess

* bump-my-version version: 0.17.0
* Python version: 3.11.5
* Operating System: MacOS Sonoma

### Description

Situation is the same as in the issue #108 - crash after updating the listed file, but before making a commit.
Downgrading back to version 0.16.2 resolves it.

### What I Did

```
Traceback (most recent call last):
  File ".../bin/bump-my-version", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File ".../site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/rich_click/rich_command.py", line 126, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File ".../site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../site-packages/bumpversion/cli.py", line 315, in bump
    do_bump(version_part, new_version, config, found_config_file, dry_run)
  File ".../site-packages/bumpversion/bump.py", line 104, in do_bump
    commit_and_tag(config, config_file, configured_files, ctx, dry_run)
  File ".../site-packages/bumpversion/bump.py", line 134, in commit_and_tag
    config.scm_info.tool.commit_to_scm(list(commit_files), config, ctx, extra_args, dry_run)
  File ".../site-packages/bumpversion/scm.py", line 169, in commit_to_scm
    cls.commit(
  File ".../site-packages/bumpversion/scm.py", line 68, in commit
    subprocess.run(cmd, env=env, capture_output=True, check=True)  # noqa: S603
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.pyenv/versions/3.11.5/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.pyenv/versions/3.11.5/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File ".../.pyenv/versions/3.11.5/lib/python3.11/subprocess.py", line 1870, in _execute_child
    env_list.append(k + b'=' + os.fsencode(v))
                               ^^^^^^^^^^^^^^
  File "<frozen os>", line 812, in fsencode
TypeError: expected str, bytes or os.PathLike object, not NoneType
make: *** [bump_patch] Error 1
```

--- a/bumpversion/scm.py
+++ b/bumpversion/scm.py
@@ -14,7 +14,7 @@ from bumpversion.utils import extract_regex_flags
 if TYPE_CHECKING:  # pragma: no-coverage
     from bumpversion.config import Config
 
-from bumpversion.exceptions import DirtyWorkingDirectoryError, SignedTagsError
+from bumpversion.exceptions import BumpVersionError, DirtyWorkingDirectoryError, SignedTagsError
 
 logger = get_indented_logger(__name__)
 
@@ -54,6 +54,12 @@ class SourceCodeManager:
     def commit(cls, message: str, current_version: str, new_version: str, extra_args: Optional[list] = None) -> None:
         """Commit the changes."""
         extra_args = extra_args or []
+        if not current_version:
+            logger.warning("No current version given, using an empty string.")
+            current_version = ""
+        if not new_version:
+            logger.warning("No new version given, using an empty string.")
+            new_version = ""
 
         with NamedTemporaryFile("wb", delete=False) as f:
             f.write(message.encode("utf-8"))
@@ -66,10 +72,13 @@ class SourceCodeManager:
         try:
             cmd = [*cls._COMMIT_COMMAND, f.name, *extra_args]
             subprocess.run(cmd, env=env, capture_output=True, check=True)  # noqa: S603
-        except subprocess.CalledProcessError as exc:  # pragma: no-coverage
-            err_msg = f"Failed to run {exc.cmd}: return code {exc.returncode}, output: {exc.output}"
+        except (subprocess.CalledProcessError, TypeError) as exc:  # pragma: no-coverage
+            if isinstance(exc, TypeError):
+                err_msg = f"Failed to run {cls._COMMIT_COMMAND}: {exc}"
+            else:
+                err_msg = f"Failed to run {exc.cmd}: return code {exc.returncode}, output: {exc.output}"
             logger.exception(err_msg)
-            raise exc
+            raise BumpVersionError(err_msg) from exc
         finally:
             os.unlink(f.name)
 

--- a/bumpversion/utils.py
+++ b/bumpversion/utils.py
@@ -75,7 +75,7 @@ def get_context(
 ) -> ChainMap:
     """Return the context for rendering messages and tags."""
     ctx = base_context(config.scm_info)
-    ctx.new_child({"current_version": config.current_version})
+    ctx = ctx.new_child({"current_version": config.current_version})
     if current_version:
         ctx = ctx.new_child({f"current_{part}": current_version[part].value for part in current_version})
     if new_version:

